llmfoundry/data/finetuning/dataloader.py

```diff
@@ -275,12 +299,43 @@ def _validate_config(dataset_cfg: DictConfig) -> None:
             'dataset, but both were None.'
         )
 
+def _downloaded_datasets_dir() -> str:
+    return os.path.join(
+        os.path.dirname(
+            os.path.dirname(os.path.dirname(os.path.realpath(__file__)))),
+        'downloaded_finetuning')
+
+def _safe_download_hf_dataset(hf_name: str, token: Union[bool, str, None]) -> str:
+    local_dataset_dir = os.path.join(_downloaded_datasets_dir(), hf_name)
+    if len(os.listdir(local_dataset_dir)) > 0:
```

Comment:
Wondering, is `> 0` a right check? is there a case where the process still downloading the file in the background where the downloading file named as a temporary file and once it is downloaded, it's get replaced with the actual file name. I have seen this with boto3 and OCI and not sure about how HF managed the file during downloading?

Reply:
I think HF will not download in the background. When the call to snapshot download is done, it will be done downloading.